scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala

```diff
@@ -43,6 +43,24 @@ class FormatWriter(formatOps: FormatOps) {
                 .getOrElse(token.syntax, token.syntax)
             sb.append(rewrittenToken)
         }
+
+        if (runner.dialect.allowTrailingCommas &&
+          state.splits.last.modification.isNewline &&
+          (formatToken.right.is[Token.RightParen] ||
```

Comment:
Conventionally we pattern match against `case FormatToken(_: Token.Comma, _: Token.RightParen | _: Token.RightBracket, _)`

Reply:
Actually, pattern matching might not be a good idea since we'll have to handle comments as well. This is fine 👍